### PR TITLE
[CPU] [WA] Use config to enable brgconv f32 kernel

### PR DESCRIPTION
**Pull request by luo-cheng2021**
### Details:
 - *Use config 'ENFORCE_EXPERIMENT_CONVOLUTION' to enable brgconv f32 for avx512 f32. Relevant PR #11866*
 - *...*

### Tickets:
- 86301

### CODE REVIEW

File: src/plugins/intel_cpu/src/config.cpp [hunk 1]
```diff
@@ -120,6 +120,19 @@ void Config::readProperties(const std::map<std::string, std::string> &prop) {
                 IE_THROW() << "Wrong value for property key " << PluginConfigParams::KEY_ENFORCE_BF16
                     << ". Expected only YES/NO";
             }
+        } else if (key == PluginConfigParams::KEY_CPU_EXPERIMENTAL) {
+            if (val == cpu_experimental_brgconv) {
+                if (dnnl::impl::cpu::x64::mayiuse(dnnl::impl::cpu::x64::avx512_core)) {
+                    enforceExperimentConv = true;
+                } else {
+                    IE_THROW() << "Platform doesn't support brgconv algorithm";
+                }
```
[the remark this answers is not in the document]
Comment (luo-cheng2021) [reply src/plugins/intel_cpu/src/config.cpp:135]: We'd better to tell the user parameter is error which may result confuse condition.

File: src/plugins/intel_cpu/src/config.cpp [hunk 2]
```diff
@@ -120,6 +120,19 @@ void Config::readProperties(const std::map<std::string, std::string> &prop) {
...
+            } else if (val.empty()) {
+                enforceExperimentConv = false;
+            } else {
+                IE_THROW() << "Wrong value for property key " << PluginConfigParams::KEY_CPU_EXPERIMENTAL
+                    << ". Expected only " << cpu_experimental_brgconv;
+            }
```
[the remark this answers is not in the document]
Comment (luo-cheng2021) [reply src/plugins/intel_cpu/src/config.cpp:137]: Tell the fail reason should be better.

File: src/plugins/intel_cpu/src/nodes/conv.cpp [hunk 3]
```diff
@@ -502,8 +548,7 @@ void Convolution::getSupportedDescriptors() {
             createDescriptor({ in_candidate }, { out_candidate });
 
             if ((inputDataType != memory::data_type::bf16 && isNspcAvailable()) ||
-                    (one_of(inputDataType, memory::data_type::bf16) &&
-                    impl::cpu::x64::mayiuse(impl::cpu::x64::avx512_core_amx))) {
+                    (acceptedFormat && impl::cpu::x64::mayiuse(impl::cpu::x64::avx512_core))) {
```
Comment (usstq) [inline src/plugins/intel_cpu/src/nodes/conv.cpp:551]: `acceptedFormat && impl::cpu::x64::mayiuse(impl::cpu::x64::avx512_core)` seems to be redundant?

Comment (luo-cheng2021) [reply src/plugins/intel_cpu/src/nodes/conv.cpp]: Removed.

**Review by usstq — APPROVED** [review]

**Review by usstq — APPROVED** [review]

**Pull request merged**